planner/filter/copy_field.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+package filter
+
+import (
+	"github.com/sourcenetwork/defradb/connor"
+	"github.com/sourcenetwork/defradb/planner/mapper"
+)
+
+// CopyField copies the given field from the provided filter.
+// The result filter preserves the structure of the original filter.
+func CopyField(filter *mapper.Filter, field mapper.Field) *mapper.Filter {
```

Comment:
suggestion: This looks like it could be made private?

Reply:
This is actually a question: what makes a function a good candidate for being public? Is it just usage from another package? Some function are just nice utilities that can be used later.